### ISSUE TITLE
[enhancement] Method to delete a row in python

### ISSUE THREAD

**Issue by Sergeileduc** [simonw/sqlite-utils#62]
Hi !
Thanks for the lib !

Obviously, every possible sql queries won't have a dedicated method.

But I was thinking : a method to delete a row (I'm terrible with names, maybe `delete_where()` or something, would be useful.

I have a Database, with primary key.

For the moment, I use :

```Python3
db.conn.execute(f"DELETE FROM table WHERE key = {key_id}")
db.conn.commit()
```
to delete a row I don't need anymore, giving his primary key.

Works like a charm.

Just an idea :

```Python3
table.delete_where_pkey({'key': key_id})
```
or something (I know, I'm terrible at naming methods...).

Pros : well, no need to write SQL query.

Cons : WHERE normally allows to do many more things (operators =, <>, >, <, BETWEEN), not to mention AND, OR, etc...
Method is maybe to specific, and/or a pain to render more flexible.

Again, just a thought. Writing his own sql works too, so...

Thanks again.
See yah.

**Comment by simonw** [comment]
This is a good idea. The individual row version of it can work like the `.update(pk, ...)` method.

**Comment by simonw** [comment]
Documentation for `table.delete()`: https://github.com/simonw/sqlite-utils/blob/19073d6d972fad9d68dd74c28544cd29083f1c12/docs/python-api.rst#deleting-a-specific-record

**Comment by simonw** [comment]
Since we have `table.rows_where(where, where_args)` it makes sense to me to also support `table.delete_where(where, where_args)`.

**Comment by simonw** [comment]
Documentation for `.delete_where()`: https://github.com/simonw/sqlite-utils/blob/169ea455fc1f1d5e5b6e44cb339ba7ffa9d49c31/docs/python-api.rst#deleting-multiple-records

**Comment by simonw** [comment]
Released as 1.12.
